api/app/utils/cron.ts

```diff
@@ -1,61 +1,110 @@
+    password: REDIS_PASSWORD,
+  },
+  defaultJobOptions: { delay: getUnixTimeFromHours(EXPIRYTIMEINHOURS) }
+})
 
 // Requeue transcript if review has expired
-async function setupExpiryTimeCron(review: Review) {
-  const expiryTime =
-    new Date(review.createdAt).getTime() +
-    getUnixTimeFromHours(EXPIRYTIMEINHOURS);
-  const expiryDate = new Date(expiryTime);
-
-  const job = new CronJob(
-    expiryDate,
-    async () => {
-      try {
-        const thisReview = await Review.findByPk(review.id);
-
-        // don't requeue transcripts whose review has been merged or submitted
-        if (thisReview?.mergedAt || thisReview?.submittedAt) return;
-
-        // TODO: archivedAt field for expired reviews
-        // const now = new Date()
-        // Review.update({ archivedAt: now }, {
-        //   where: { id: review.id }
-        // })
-
-        await Transcript.update(
-          { status: TRANSCRIPT_STATUS.QUEUED, claimedBy: null },
-          {
-            where: { id: review.transcriptId },
-          }
-        );
-        const totalItems = await Transcript.count();
-        const limit = 5;
-        const totalPages = Math.ceil(totalItems / limit);
-        for (let page = 1; page <= totalPages; page++) {
-          await deleteCache(`transcripts:page:${page}`);
-        }
-        await deleteCache(`transcript:${review.transcriptId}`);
-        await redis.srem("cachedTranscripts", review.transcriptId);
-      } catch (err) {
-        console.log(err);
-      }
-    },
-    null,
-    false,
-    undefined
-  );
-  job.start();
-  return;
+export const addToExpiryQueue = async (reviewId: number) => {
+  await expiryQueue.add({reviewId})
 }
 
-export { setupExpiryTimeCron };
+expiryQueue.process(async (job, done) => {
+  const { reviewId } = job.data
+  try {
+    const thisReview = await Review.findByPk(reviewId);
+
+    if (!thisReview) return;
+
+    // don't requeue transcripts whose review has been merged or submitted
+    if (thisReview?.mergedAt || thisReview?.submittedAt) return;
+
+    const now = new Date()
+    thisReview.update({ archivedAt: now })
+
+    await Transcript.update(
+      { status: TranscriptStatus.queued, claimedBy: null },
+      {
+        where: { id: thisReview.transcriptId },
+      }
+    );
+    console.log("updated transcript and review")
+    await resetRedisCachedPages();
+    await deleteCache(`transcript:${thisReview.transcriptId}`);
+    await redis.srem("cachedTranscripts", thisReview.transcriptId);
+
+  } catch (err) {
+    console.log(err);
+  }
+  done()
+})
+
+
+// run CRON once daily to archive expired review that could have slipped by
+
+const dailyCheckForMissedExpiredReviews = new Queue("daily-cron-expired", {
+  redis: {
+    port: REDIS_PORT,
+    host: REDIS_HOST,
+    password: REDIS_PASSWORD,
+  },
+  defaultJobOptions: { repeat: { cron: "0 0 * * *" } },
+})
+
+export const startDailyExpiredReviewsCheck = async () => await dailyCheckForMissedExpiredReviews.add({})
```

Comment:
Where would we call this? I can't find where you're using this function

Reply:
In the same file at the end, or in `server.ts` after sequelize sync. Bull is smart enough not to add the same instance of a repeatable job to a queue.
So I didn't call this on purpose for a reason. I'll drop it on discord

Reply:
Okay. But why aren't you calling the function?

Reply:
its on discord boss 👀
TLDR; small chance we clear `claimedby` and `queuestatus` of a transcript that got claimed after it was requeued using the previous implementation. Then again we don't know when the previous implementation stopped working, so it's more or less a non issue

I'm calling it now tho